### PR TITLE
fontforge-gtk: Add support for libspiro.

### PR DESCRIPTION
**Pull request by lprndn**
###### Motivation for this change
`fontforge-gtk` is mainly for font editing. Adding Spiro tools let us have a complete toolkit.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libspiro/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{stdenv, autoreconfHook, autoconf, pkgconfig, fetchgit}:
+
+stdenv.mkDerivation rec {
+  name = "libspiro";
+  version = "20170128";
+  src = fetchgit {
```
Comment (joachifm) [inline pkgs/development/libraries/libspiro/default.nix]: Does this rely on submodules? If not, source archives are strongly preferred. You can use `fetchFromGitHub` for convenience.

**Comment by lprndn** [comment]
Switched to stable release. Therefore used `fetchurl` to fetch sources.

File: pkgs/development/libraries/libspiro/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  name = "libspiro-${version}";
+  version = "0.5.20150702";
+  src = fetchurl {
+    url = "https://github.com/fontforge/libspiro/releases/download/${version}/${name}.tar.gz";
+    sha256 = "0z4zpxd3nwwchqdsbmmjbp13aw5jg8v5p1993190bpykkrjlh6nv";
+  };
+
+  nativeBuildInputs = [pkgconfig];
+
+  meta = with stdenv.libs; {
```
Comment (joachifm) [inline pkgs/development/libraries/libspiro/default.nix]: `stdenv.libs` should be `stdenv.lib`.

**Comment by joachifm** [comment]
The travis build error looks legit

**Comment by lprndn** [comment]
@joachifm `pdf2htmlEX`, which use `fontforge-gtk` as  a dependency, doesn't seem to build with the addition of libspiro. I'm just asking myself if it's really necessary to use `fontforge-gtk` instead of `fontforge` there. To my knowledge, it's a CLI tool and, after testing, it actually builds without problems with standard fontforge. What do you think?

**Comment by joachifm** [comment]
cc @taktoa may know more about `pdf2htmlEx`.

**Comment by taktoa** [comment]
pdf2htmlEX should be fine with `fontforge`; I think I had suspected that some features would be enabled by switching to `-gtk` and then forgot to switch it back (I packaged it a while ago, so I don't remember my exact thought process).

**Comment by joachifm** [comment]
@lprndn this needs a rebase

**Comment by lprndn** [comment]
Rebased. Kept 2 commits, I think it might be nice to leave pdef2htmlEX modifications on their own.

**Comment by lprndn** [comment]
Since 17.03 release rush is over, is there any blocking issue to merge this commit?

**Comment by vcunat** [comment]
I see a merge conflict, though it might be something trivial.

**Comment by lprndn** [comment]
Merge conflict comes from pdf2htmlEX modifications discussed before. The maintainer of the package seems to agree for the changes.

**Comment by taktoa** [comment]
Does this mean I need to make a separate pull request for the pdf2htmlEX changes or will this PR include those changes? In other words, do you need me to take any action?

**Comment by lprndn** [comment]
I made the changes myself in this commit so I don't think you need to worry. Conflict is simple so it can be resolved directly in github.

**Comment by vcunat** [comment]
Yes, it's also possible to do it locally, e.g. the same way (merge master into your branch).

**Pull request merged**